src/common/ipc/IPC.ts

```diff
@@ -0,0 +1,113 @@
+import { FailableAsync } from '../utils/failable'
+import { AUTH_TAGGED_EVENTS } from './model/authentication'
+import { BITCOIND_TAGGED_EVENTS } from './model/bitcoind'
+import { DLC_TAGGED_EVENTS } from './model/dlc'
+import { FILE_TAGGED_EVENTS } from './model/file'
+import { ORACLE_TAGGED_EVENTS } from './model/oracle'
+import { UPDATE_TAGGED_EVENTS } from './model/update'
+import { USER_TAGGED_EVENTS } from './model/user'
+
+export interface EventIPCHandler {
+  register(
+    event: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    callback: (data: any, ...args: any[]) => unknown
+  ): () => void
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  call(event: string, data: unknown, ...args: any): Promise<unknown>
+}
+
+// The IPCHandler should be set at start up
+export let IPCHandler: EventIPCHandler
+
+export function setEventIPCHandler(handler: EventIPCHandler): void {
+  IPCHandler = handler
+}
+
+export function ipcCallRenderer<T extends ANY_TAGS>(
```

Comment:
I'm not a big fan of having this in common. I feel this should be in Browser (same for `ipcCallMain` should only be in renderer).

Reply:
In general I feel things are a bit overcomplicated due to trying to handle both browser and renderer in the same class/functions. I see it could be nice to have, but I would rather have things a bit more cleanly separated and a bit more verbose I think.

Reply:
It's also weird to have a singleton in common. I guess it means that it gets instantiated in both, but I don't find that clean. I would rather have the utility functions take it as parameter.